odl/deform/mass_preserving.py

```diff
@@ -0,0 +1,548 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators and functions for general and mass-preserving deformation."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.discr import DiscreteLp, DiscreteLpVector
+from odl.operator.operator import Operator
+from odl.space import ProductSpaceVector, ProductSpace
+
+
+__all__ = ('GeneralDeformFixedTempl', 'GeneralDeformFixedDeform',
+           'MassPresvDeformFixedTempl', 'MassPresvDeformFixedDeform')
+
+
+def _general_deform(template, deformation, out=None):
```

Comment:
This looks *very* similar to the "same" method in deform/linear. What is different? Could they perhaps be merged to a single function?